assets/src/components/customers/CustomerDetailsSidebar.tsx

```diff
@@ -0,0 +1,316 @@
+            name="Timezone"
+            value={getTimezoneValue(timezone)}
+          />
+          <CustomerDetailsProperty
+            Icon={DesktopOutlined}
+            name="Browser"
+            value={browser}
+          />
+          <CustomerDetailsProperty
+            Icon={DesktopOutlined}
+            name="OS"
+            value={os}
+          />
+          <CustomerDetailsProperty
+            Icon={DesktopOutlined}
+            name="IP"
+            value={lastIpAddress}
+          />
+        </CustomerDetailsSection>
+
+        <Divider dashed />
+
+        <CompanyDetailsSection company={company} />
+
+        <Divider dashed />
+
+        {hasMetadata && (
+          <>
+            <CustomerDetailsSection title="Metadata">
+              {Object.entries(metadata).map(([key, value]) => (
+                <CustomerDetailsProperty
+                  Icon={InfoCircleOutlined}
+                  key={key}
+                  name={key}
+                  value={value}
+                />
+              ))}
+            </CustomerDetailsSection>
+            <Divider dashed />
+          </>
+        )}
+
+        <CustomerDetailsSection title="Tags">
+          <SidebarCustomerTags customerId={customerId} />
+        </CustomerDetailsSection>
+      </Box>
+    </CustomerDetailsCard>
+  );
+};
+
+const CompanyDetailsSection = ({company}: {company: Company | undefined}) => {
+  let body;
+
+  if (company) {
+    const {
+      id: companyId,
+      name,
+      website_url: websiteUrl,
+      slack_channel_id: slackChannelId,
+      slack_channel_name: slackChannelName,
+    } = company;
+
+    body = (
+      <>
+        <CustomerDetailsProperty
+          Icon={TeamOutlined}
+          name="Name"
+          value={<Link to={`/companies/${companyId}`}>{name}</Link>}
+        />
+        <CustomerDetailsProperty
+          Icon={LinkOutlined}
+          name="Website"
+          value={websiteUrl}
+        />
+        <CustomerDetailsProperty
+          Icon={LinkOutlined}
+          name="Slack Channel"
+          value={
+            slackChannelId &&
+            slackChannelName && (
+              <a
+                href={`https://slack.com/app_redirect?channel=${slackChannelId}`}
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                {slackChannelName}
+              </a>
+            )
+          }
+        />
+      </>
+    );
+  } else {
+    // TODO: add ability to link customer to company
+    body = <Text>Customer is not linked to a company.</Text>;
+  }
+
+  return (
+    <CustomerDetailsSection title="Company">{body}</CustomerDetailsSection>
+  );
+};
+
+export const CustomerDetailsSection = ({
+  title,
+  children,
+}: {
+  title: string;
+  children: React.ReactNode;
+}) => {
+  return (
+    <Box>
+      <Box mb={2}>
+        <Text strong>{title}</Text>
+      </Box>
+      {children}
+    </Box>
+  );
+};
+
+const CustomerDetailsProperty = ({
+  Icon,
```

Comment:
nit -- i think react props should always be lowercase? 🤔 

but i see this is passing in the icon class/function rather than the component itself, so i see why it's capitalized... i think i'd rather pass in the component itself even if we need to explicitly set the color every time, e.g.
```jsx
<CustomerDetailsProperty
  icon={<LinkOutlined style={{color: colors.primary}} />}
  name="Website"
  value={websiteUrl}
/>
```

Reply:
It's been a hot minute so I forgot that that was a best practice 😅 It didn't seem to be a problem, but I'm happy to change it to lowercase. 

Thoughts on a helper method used to reduce duplication? 

```
const getIconForComponent = (Icon) => (<Icon style={{color: color.primary}}>);
```

Reply:
down for that! :+1: